pyke/kepclean.py

```diff
@@ -0,0 +1,161 @@
+
+    Examples
+    --------
+    .. code-block:: bash
+
+        $ kepclean kplr002436324-2009259160929_llc.fits
+          --verbose --overwrite
+    """
+
+    if outfile is None:
+        outfile = infile.split('.')[0] + "-{}.fits".format(__all__[0])
+
+    # log the call
+    hashline = '--------------------------------------------------------------'
+    kepmsg.log(logfile, hashline, verbose)
+    call = ('KEPCLEAN -- '
+            + 'infile={}'.format(infile)
+            + ' outfile={}'.format(outfile)
+            + ' zero={}'.format(zero)
+            + ' overwrite={}'.format(overwrite)
+            + ' verbose={}'.format(verbose)
+            + ' logfile={}'.format(logfile))
+    kepmsg.log(logfile, call+'\n', verbose)
+
+    # start time
+    kepmsg.clock('KEPCLEAN started at',logfile,verbose)
+
+    # overwrite output file
+    if overwrite:
+        kepio.overwrite(outfile, logfile, verbose)
+    if kepio.fileexists(outfile):
+        errmsg = 'ERROR -- KEPCLIP: ' + outfile + ' exists. Use --overwrite'
+        kepmsg.err(logfile, errmsg, verbose)
+
+
+    # open input file
+    instr = pyfits.open(infile, 'readonly')
+    tstart, tstop, bjdref, cadence = kepio.timekeys(instr, infile, logfile,
+                                                    verbose)
+
+    #loop over all extentions in the file
+    for ext in range(len(instr)):
+        try:
+            kepmsg.log(logfile, 'KEPCLEAN - Cleaning extention {} ({})'.format(ext,instr[ext].header['EXTNAME']), verbose)
+        except:
+            kepmsg.log(logfile, 'KEPCLEAN - Cleaning extention {}'.format(ext), verbose)
+        #Dig out the header names that are data and find those that are 'FLUX'
+        hnames = []
+        flux_keys = []
+        for k in (instr[ext].header):
+            if k.startswith('TTYPE'):
+                hname = instr[ext].header[k]
+                try:
+                    dat = instr[ext].data[instr[ext].header[k]]
+                    hnames.append(hname)
+                    if 'FLUX' in hname:
+                        flux_keys.append(True)
+                    else:
+                        flux_keys.append(False)
+                except:
+                    continue
+        flux_keys = np.asarray(flux_keys,dtype=bool)
+        if np.any([(len(flux_keys) == 0),(len(np.where(flux_keys==True)[0]) == 0)]) :
+            continue
+        hnames = np.asarray(hnames)
+        #if there are no FLUX keys to correct by, just move on
+        kepmsg.log(logfile, 'KEPCLEAN - Found flux keys {}'.format(hnames[flux_keys]), verbose)
+
+        #Loop through and remove indicies where any 'FLUX' keys have NaNs
+        length = len(instr[ext].data[hnames[flux_keys][0]])
+        fin = set(list(np.arange(length)))
+        for y in hnames[flux_keys]:
+            ydat = instr[ext].data[y]
+            if len(np.shape(ydat)) > 1:
+                for i in range(len(np.shape(ydat))-1):
+                    bad = np.all(np.isfinite(ydat)==False,axis=-1)
```

Comment:
``np.isfinite(ydat)==False`` this is not recommend. Instead do, ``np.isfinite(ydat) is False``.

Reply:
I think you could even do ``bad = np.all(~np.isfinite(ydat),axis=-1)``?

Reply:
Reference: https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions